packages/synapse-react-client/src/lib/containers/AccessRequirementList/AccessRequirementList.tsx

```diff
@@ -141,12 +146,19 @@ export default function AccessRequirementList(
 
   const entityInformation = useGetInfoFromIds<EntityHeader>(entityHeaderProps)
 
-  const { data: fetchedRequirements } = useGetAccessRequirementsForEntity(
-    entityId,
-    {
-      enabled: Boolean(!accessRequirementFromProps && entityId),
-    },
-  )
+  const determineFetchedARs = () => {
+    const fetched = teamId
+      ? useGetAccessRequirementsForTeam(teamId, {
+          enabled: Boolean(!accessRequirementFromProps && teamId),
+        })
+      : useGetAccessRequirementsForEntity(entityId, {
+          enabled: Boolean(!accessRequirementFromProps && !teamId && entityId),
+        })
+
+    return fetched || {}
+  }
+
+  const { data: fetchedRequirements } = determineFetchedARs()
```

Comment:
This is breaking the rules of hooks (indicated in the lint job failure).

Maybe you could do something like this?

```suggestion
  const { data: fetchedRequirementsForTeam } = useGetAccessRequirementsForTeam(
    entityId,
    {
      enabled: Boolean(!accessRequirementFromProps && teamId),
    },
  )
  const { data: fetchedRequirementsForEntity } = useGetAccessRequirementsForEntity(
    entityId,
    {
      enabled: Boolean(!accessRequirementFromProps && entityId && !teamId),
    },
  )

  const fetchedRequirements = teamId ? fetchedRequirementsForTeam : fetchedRequirementsForEntity
```

Reply:
Good catch, thank you. I had similar to suggested originally but shortened things, only to realize my linter isn't catching correctly (Intellij IDEA). I forced a bad use of `useMemo` with a hook inside as well to test but there seems to be some inconsistencies in warnings.

Do you happen to be pointing your IDE to a specific eslint config within `synapse-react-client` or letting it automatically scan the project?

Reply:
Sorry I meant within `synapse-web-monorepo`